Simulator/Layouts/Layout.cpp

```diff
@@ -58,6 +58,11 @@ void Layout::registerGraphProperties()
    // initial graph
 }
 
+void Layout::printLayout()
+{
+   // Empty implementation because not all Layouts implement it but it is used by the Model
+}
+
```

Comment:
Can't this be pure virtual in this class?

Reply:
> Can't this be pure virtual in this class?

Currently, the code in this method is never executed because the call in `Simulator::simulate()`, where it was being used for debugging, is commented out:

```
   // DEBUG_MID(model->logSimStep();) // Generic model debug call
```

The code seems to print a formatted grid of the neurons' layout to standard output, so I moved it to `FixedLayout` but kept it as a call to `Layout` in `Model` without down-casting. And, since no other Layout implements this I had to add an empty implementation in the base class.

This is a hack to keep a block of code that is not even being used, so I don't know if it is worth it. I believe this was used at some point for debugging the `Connection Growth` simulation model but is now dead code left behind. We could remove the `Model::logSimStep()` method and leave the piece code snippet in `FixedLayout` only, in case anyone ever needs to use it.

Reply:
OK, I say comment out all of this dead code, with a brief comment.

Reply:
I commented out the dead code in the Model class with a comment, removed the method from the Layout base class, and put a comment in the FixedLayout class where I moved the original code snippet, along with the `width_` and `height_` variables.